### DISCUSSION TITLE
How does the idling client connection get closed automatically?

### DISCUSSION THREAD

**Discussion by y2241** [redis/lettuce#2280] [General]
I noticed that when my program (jvm) exists, my redis connection (I am not using redis connection pool) is closed, even I do not explicitly close the connection nor the client itself. The redis sever (aws elastic cache has timeout 0, means it keeps the client connection forever unless client decides to close  the connection).
In aws lambda function, I noticed similar behavior, except that the execution environment may stay longer, so the connection is not immediately closed but is closed eventually.

I wonder what mechanism the Lettuce uses to close idling connection?
Thanks.

**Comment by mp911de** [comment]
Lettuce does not close idle connections. If the JVM terminates, then so do the TCP connections. Ideally, you close a connection if you no longer require it.
